organization/src/integrationTest/kotlin/com/cosmotech/organization/service/OrganizationServiceIntegrationTest.kt

```diff
@@ -1283,1447 +1283,1502 @@ class OrganizationServiceIntegrationTest : CsmRedisTestBase() {
-                  createTestOrganizationWithSimpleSecurity(
-                      name, TEST_ADMIN_USER_ID, ROLE_NONE, ROLE_NONE))
-          runAsPlatformAdmin()
-          assertNotNull(organizationApiService.getOrganizationSecurity(organizationRegistered.id!!))
-        }
+    @Test
+    fun `setOrganizationDefaultSecurity with owned organization and non-existing role`() {
+      assertThrows<CsmClientException> {
+        val name = "o-connector-test-1"
+        val organizationRegistered =
+            organizationApiService.registerOrganization(createTestOrganization(name))
+        organizationApiService.setOrganizationDefaultSecurity(
+            organizationRegistered.id!!, OrganizationRole(UNKNOWN_IDENTIFIER))
       }
+    }
 
-      @Test
-      fun `setOrganizationDefaultSecurity with owned organization and existing role`() {
-        assertDoesNotThrow {
-          val name = "o-connector-test-1"
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
-          val defaultRoleCreated = organizationRegistered.security?.default
-          assertNotNull(
-              organizationApiService.setOrganizationDefaultSecurity(
-                  organizationRegistered.id!!, OrganizationRole(ROLE_ADMIN)))
-          val defaultRoleUpdated =
-              organizationApiService.getOrganizationSecurity(organizationRegistered.id!!).default
-          assertNotEquals(defaultRoleCreated, defaultRoleUpdated)
-        }
+    @Test
+    fun `setOrganizationDefaultSecurity with not owned organization`() {
+      assertDoesNotThrow {
+        val name = "o-connector-test-1"
+        runAsDifferentOrganizationUser()
+        val organizationRegistered =
+            organizationApiService.registerOrganization(createTestOrganization(name))
+        val defaultRoleCreated = organizationRegistered.security?.default
+        runAsPlatformAdmin()
+        assertNotNull(
+            organizationApiService.setOrganizationDefaultSecurity(
+                organizationRegistered.id!!, OrganizationRole(ROLE_ADMIN)))
+        val defaultRoleUpdated =
+            organizationApiService.getOrganizationSecurity(organizationRegistered.id!!).default
+        assertNotEquals(defaultRoleCreated, defaultRoleUpdated)
       }
+    }
 
-      @Test
-      fun `setOrganizationDefaultSecurity with owned organization and non-existing role`() {
-        assertThrows<CsmClientException> {
-          val name = "o-connector-test-1"
-          val organizationRegistered =
-              organizationApiService.registerOrganization(createTestOrganization(name))
-          organizationApiService.setOrganizationDefaultSecurity(
-              organizationRegistered.id!!, OrganizationRole(UNKNOWN_IDENTIFIER))
-        }
+    @Test
+    fun `setOrganizationDefaultSecurity with not owned organization and no WRITE_SECURITY_PERMISSION`() {
+      assertDoesNotThrow {
```

Comment:
assertDoesNotThrow should be specific else it's the same as not having it

Reply:
same comment as above